chia-bls/src/public_key.rs

```diff
@@ -175,6 +286,25 @@ impl DerivableKey for PublicKey {
     }
 }
 
+impl FromClvm for PublicKey {
+    fn from_clvm(a: &Allocator, ptr: NodePtr) -> clvm_traits::Result<Self> {
+        Self::from_bytes(
+            &node_to_bytes(a, ptr)
```

Comment:
I don't think this is what you want here, `node_to_bytes` is the serialized CLVM program rather than the atom, which means it includes length information. You need to do:

```rust
if let SExp::Atom = a.sexp(ptr) {
    // a.atom(ptr)
} else {
    Err(clvm_traits::Error::ExpectedAtom)
}
```

Reply:
I'm not sure what I was thinking there..